runbenchmarks.py

```diff
@@ -89,13 +91,38 @@ def invoke_lambda(configs: dict, config_file: str):
     payload = {
         "config_file": config_file
     }
-    lambda_client.invoke(
+    response = lambda_client.invoke(
         FunctionName=configs["LAMBDA_FUNCTION_NAME"],
         InvocationType='RequestResponse',
         Payload=json.dumps(payload)
     )
     logger.info("AWS Batch jobs submitted by %s.", configs["LAMBDA_FUNCTION_NAME"])
+    
+    job_ids = json.loads(response['Payload'].read().decode('utf-8'))['job_ids']
+    return job_ids
+
+def wait_for_jobs_to_complete(batch_client, job_ids: list):
+    logger.info("Waiting for jobs to complete...")
+    while True:
+        all_jobs_completed = True
+        failed_jobs = []
+
+        for job_id in job_ids:
+            response = batch_client.describe_jobs(jobs=[job_id])
+            job = response["jobs"][0]
+            job_status = job["status"]
+
+            if job_status == "FAILED":
+                failed_jobs.append(job_id)
+            elif job_status not in ["SUCCEEDED", "FAILED"]:
```

Comment:
There are other states like "CANCELLED". Are we sure these are the only 2 states that should be marked as complete?

Reply:
Here is a [list](https://docs.aws.amazon.com/batch/latest/userguide/job_states.html) of all available states for AWS Batch jobs, among which the final states are `SUCCEEDED` and `FAILED`. If a job is cancelled manually, it goes to `FAILED`.